internal/cloud/backend_plan.go

```diff
@@ -373,6 +353,128 @@ in order to capture the filesystem context the remote workspace expects:
 	return r, nil
 }
 
+// renderPlanLogs reads the streamed plan JSON logs and calls the JSON Plan renderer (jsonformat.RenderPlan) to
+// render the plan output. The plan output is fetched from the redacted output endpoint.
+func (b *Cloud) renderPlanLogs(ctx context.Context, op *backend.Operation, run *tfe.Run) error {
+	logs, err := b.client.Plans.Logs(ctx, run.Plan.ID)
+	if err != nil {
+		return err
+	}
+
+	reader := bufio.NewReaderSize(logs, 64*1024)
+
+	// Since the plan output is not read from the streamed logs and instead fetched
+	// from the redacted output endpoint, we need to defer any logs such that the
+	// plan output is rendered *first* before we write any subsequent logs after the plan output.
+	deferredLogs := []*jsonformat.JSONLog{}
+
+	// This variable helps us track the first instance we encounter a log of type plan_output.
+	// When the plan has started, we should be deferring subsequent logs until after the plan
+	// has been rendered.
+	planStarted := false
+
+	if b.CLI != nil {
+		for next := true; next; {
+			var l, line []byte
+			var err error
+
+			for isPrefix := true; isPrefix; {
+				l, isPrefix, err = reader.ReadLine()
+				if err != nil {
+					if err != io.EOF {
+						return generalError("Failed to read logs", err)
+					}
+					next = false
+				}
+
+				line = append(line, l...)
+			}
+
+			if next || len(line) > 0 {
+				log := &jsonformat.JSONLog{}
+				if err := json.Unmarshal(line, log); err != nil || log == nil {
+					// If we can not parse the line as JSON, we will simply
+					// print the line. This maintains backwards compatibility for
+					// users who do not wish to enable structured output in their
+					// workspace.
+					b.CLI.Output(b.Colorize().Color(string(line)))
```

Comment:
Same note about colorization as above. We should never be receiving color codes (e.g. `[red]`) from a cloud text stream, only plain text with possible ANSI color codes.